core/embed/unix/main.c

```diff
@@ -61,6 +61,8 @@
 STATIC bool compile_only = false;
 STATIC uint emit_opt = MP_EMIT_OPT_NONE;
 
+STATIC bool mp_initialized = false;
+
```

Comment:
To keep it as simple as possible: Since you have removed `main_clean_exit()` from `trezor_shutdown()`, this flag  is no longer needed. Additionally, I would suggest adding the static keyword to `main_clean_exit()`.

Reply:
ade1fdcc22b2ce78d8faf02864e52cebeb997ce3

Static won't work because it's called from another file. I could move it there but then I'd have to #include bunch of micropython stuff where there's currently none. At least the declaration could be moved to a header file, but which one ...

Reply:
Ah, okay. This would require deeper plowing... So I am satisfied.